keyboards/adpenrose/kintsugi/keymaps/franky/keymap.c

```diff
@@ -0,0 +1,173 @@
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+    if (clockwise){
+        /* Check if left shift is pressed: */
+        if (selected_layer < 3 && get_mods() & MOD_BIT(KC_LSFT)){
+            selected_layer ++;
+            layer_move(selected_layer); /* Jump up one layer. */
+        } else {
+            /* If shift isn't pressed, encoder will do this stuff: */
+            switch (get_highest_layer(layer_state)){
+                case 3:
+                    tap_code(KC_MNXT);
+                    break;
+                default:
+                    tap_code(KC_VOLU);
+                    break;
+            }
+        }
+    } else {
+        /* Check if left shift is pressed: */
+        if (selected_layer > 0 && get_mods() & MOD_BIT(KC_LSFT)){
+            selected_layer --;
+            layer_move(selected_layer); /* Go down one layer. */
+        } else {
+            /* If shift isn't pressed, encoder will do this stuff: */
+            switch (get_highest_layer(layer_state)){
+                case 3:
+                    tap_code(KC_MPRV);
+                    break;
+                default:
+                    tap_code(KC_VOLD);
+                    break;
+            }
+        }
+    }
+    return true;
+}
+#endif
+
+/* Rotation of the OLED: */
+#ifdef OLED_ENABLE
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_270;
+}
+
+/* Function that renders the kintsugi logo in the desired order. */
+static void render_logo(void) {
+    static const char PROGMEM logo_1[] = {
+        0x83, 0x84, 0x85, 0x86, 0x87, 0xA3, 0xA4, 0xA5, 0xA6, 0xA7, 0xC3, 0xC4, 0xC5, 0xC6, 0xC7, 0x00
+    };
+    static const char PROGMEM logo_2[] = {
+        0x88, 0x89, 0x8A, 0x8B, 0x8C, 0xA8, 0xA9, 0xAA, 0xAB, 0xAC, 0xC8, 0xC9, 0xCA, 0xCB, 0xCC, 0x00
+    };
+    static const char PROGMEM logo_3[] = {
+        0x8D, 0x8E, 0x8F, 0x90, 0x91, 0xAD, 0xAE, 0xAF, 0xB0, 0xB1, 0xCD, 0xCE, 0xCF, 0xD0, 0xD1, 0x00
+    };
+    oled_set_cursor(1,0);
+    oled_write_P(logo_1, false);
+    oled_set_cursor(1,4);
+    oled_write_P(logo_2, false);
+    oled_set_cursor(1,8);
+    oled_write_P(logo_3, false);
+}
+
+/* Function that renders the current layer to the user. */
+void render_layer(void) {
+    oled_set_cursor(0,12);
+    switch (get_highest_layer(layer_state)){
+        case 0:
+            oled_write_P(PSTR("_BASE"), false);
+            break;
+        case 1:
+            oled_write_P(PSTR("_KCAD"), false);
+            break;
+        case 2:
+            oled_write_P(PSTR("_NMPD"), false);
+            break;
+        case 3:
+            oled_write_P(PSTR("_FNCT"), false);
+            break;
+        default:
+            oled_write_P(PSTR("_OhNo"), false);
+            break;
+    }
+}
+
+/* The following function displays wpm to the user. */
+void render_wpm(void) {
+    oled_set_cursor(1,14);
+    sprintf(wpm_str, "%03d", get_current_wpm());
+    oled_write(wpm_str, false);
```

Comment:
```suggestion
    oled_write(get_u8_str(get_current_wpm(), '0'), false);
```

Reply:
I've tried using this function, but the WPMs that it displays are not accurate. Moreover, in the docs there's no indication that people should use this new code instead of the one that I used. The only advantage that the code that you are suggesting gives is the memory space, but my firmware has plenty of free space left.

Reply:
Odd that it's not accurate.  I've been using it and it appears to be plenty accurate. 

As for documentation, it's in the squeezing avr doc, but should be added to "useful functions" as well. 

And the code is very new (as of nov 27th). 

> The only advantage that the code that you are suggesting gives is the memory space, but my firmware has plenty of free space left.

My concern here is that while the default keymap may have plenty of free space ... having RGB and OLEDs enabled is still a lot of firmware usage.   And if people start adding stuff ontop of that, such as rgb light layers, bongo cats, etc... that will add up fast.

Compounded by the fact that people will just copy and paste existing code... keeping the sprintf/snprintf function calls means that a lot of people will still use it.